### DISCUSSION TITLE
I like TRMM, I really do but....

### DISCUSSION THREAD

**Discussion by ghost** [amidaware/tacticalrmm#1749] [Ideas]
Apologies in advance as this will no doubt upset some people. 

As the title says I like TRMM, I honestly do and think it's a worthwhile project.

Over the last couple of years every so often I've fired up a VM, installed and configured some of the endpoints from my existing RMM software and tried it out to see what improvements there have been and unfortunately I always seem to come away disappointed after a week or so of playing with it.

The excellent:

1. Agents are small and quick to execute.

2. I LOVE the speed that scripts execute on endpoints - it often amazes me how quickly results come back after clicking on the Run button. 

My major (and minor) gripes are:

1. Tasks are still only supported on Windows. Why not macOS and Linux yet?

3. Start up scripts. I was happy to see the onboarding feature appear and with a wee bit of tweaking it could easily become a "run script xyz whenever an endpoint checks in AKA "run scripts at start up." This one is THE major blocker for me. 

4. Scrolling through the list of endpoints is visually jarring (at least on Safari). Please make it smoother!

I'm quite happy to pay for a finished or nearly-finished product but as things stand I won't be doing yet that until I can replace my current RMM with TRMM and have some sort of feature parity. 

I'm not a programmer, simply an end user who wants to use TRMM. Make my wishes come true?

**Comment by bbrendon** [comment]
TRMM as it is right now is great for scripting and API. You can already do 1 and 2 on your list with some creativity. I honestly don't mind not having them because you can just use the existing scripting features to implement them yourself. No need for features that aren't actually needed. I can see your point though for people coming from RMMs that have TONS of features in the GUI and just want something similar, but I'm not sure the "powers that be" envision this product ever being like that.

I personally find the lack of features and simplicity with the requirement of more scripting refreshing.
